tools/fm-editor/graph/FeatureModelGraph.cpp

```diff
@@ -0,0 +1,216 @@
+                 NextChildren.begin(),
+                 [](FeatureEdge *Edge) { return Edge->targetNode(); });
+  positionRec(1, NextChildren, 5);
+  EntryNode->setPos(EntryNode->childrenWidth() / 2.0, 10);
+}
+
+void FeatureModelGraph::buildRec(FeatureNode *CurrentFeatureNode) {
+  for (auto *Feature :
+       CurrentFeatureNode->getFeature()->getChildren<vara::feature::Feature>(
+           1)) {
+    auto Node = std::make_unique<FeatureNode>(Feature);
+    auto *Edge = new FeatureEdge(CurrentFeatureNode, Node.get());
+    scene()->addItem(Edge);
+    scene()->addItem(Node.get());
+    buildRec(Node.get());
+    Nodes.push_back(std::move(Node));
+  }
+  for (auto *Relation : CurrentFeatureNode->getFeature()
+                            ->getChildren<vara::feature::Relationship>(1)) {
+    for (auto *Feature : Relation->getChildren<vara::feature::Feature>(1)) {
+      auto Node = std::make_unique<FeatureNode>(Feature);
+      auto *Edge = new FeatureEdge(CurrentFeatureNode, Node.get());
+      scene()->addItem(Edge);
+      scene()->addItem(Node.get());
+      buildRec(Node.get());
+      Nodes.push_back(std::move(Node));
+    }
+  }
+}
+
+int FeatureModelGraph::positionRec(const int CurrentDepth,
+                                   const std::vector<FeatureNode *> &Children,
+                                   const unsigned long Offset) {
+  if (Children.empty()) {
+    return CurrentDepth - 1;
+  }
+  int MaxDepth = CurrentDepth;
+  auto NextOffset = Offset;
+  for (FeatureNode *Node : Children) {
+    auto NextChildren = std::vector<FeatureNode *>(Node->children().size());
+    auto CurrentChildren = Node->children();
+    std::transform(CurrentChildren.begin(), CurrentChildren.end(),
+                   NextChildren.begin(),
+                   [](FeatureEdge *Edge) { return Edge->targetNode(); });
+    int const Depth = positionRec(CurrentDepth + 1, NextChildren, NextOffset);
+    int Width = Node->childrenWidth() + 5;
+    Node->setPos(double(NextOffset) + Width / 2.0, 100 * CurrentDepth);
+    NextOffset += Width;
+    MaxDepth = MaxDepth < Depth ? Depth : MaxDepth;
+  }
+
+  return MaxDepth;
+}
+
+void FeatureModelGraph::keyPressEvent(QKeyEvent *Event) {
+  switch (Event->key()) {
+  case Qt::Key_Plus:
+    zoomIn();
+    break;
+  case Qt::Key_Minus:
+    zoomOut();
+    break;
+  default:
+    QGraphicsView::keyPressEvent(Event);
+  }
+}
+
+#if QT_CONFIG(wheelevent)
+void FeatureModelGraph::wheelEvent(QWheelEvent *Event) {
+  scaleView(pow(2., -Event->angleDelta().y() / 240.0));
+}
+#endif
+
+void FeatureModelGraph::drawBackground(QPainter *Painter, const QRectF &Rect) {
+  Q_UNUSED(Rect)
+
+  // Shadow
+  QRectF const SceneRect = this->sceneRect();
+  QRectF const RightShadow(SceneRect.right(), SceneRect.top() + 5, 5,
+                           SceneRect.height());
+  QRectF const BottomShadow(SceneRect.left() + 5, SceneRect.bottom(),
+                            SceneRect.width(), 5);
```

Comment:
```suggestion
  const QRectF SceneRect = this->sceneRect();
  const QRectF RightShadow(SceneRect.right(), SceneRect.top() + 5, 5,
                           SceneRect.height());
  const QRectF BottomShadow(SceneRect.left() + 5, SceneRect.bottom(),
                            SceneRect.width(), 5);
```
Some how, for some const's you changed from west const to east const :thinking: